tests/worker/test_workflow.py

```diff
@@ -3977,3 +3980,69 @@ async def test_workflow_timeout_support(client: Client, approach: str):
                 found_timer = True
                 break
         assert found_timer
+
+
+@workflow.defn
+class BuildIDInfoWorkflow:
+    do_finish = False
+
+    @workflow.run
+    async def run(self):
+        await asyncio.sleep(1)
+        if workflow.info().get_current_build_id() == "1.0":
+            await workflow.execute_activity(
+                say_hello, "yo", schedule_to_close_timeout=timedelta(seconds=5)
+            )
+        await workflow.wait_condition(lambda: self.do_finish)
+
+    @workflow.query
+    def get_build_id(self) -> str:
+        return workflow.info().get_current_build_id()
+
+    @workflow.signal
+    async def finish(self):
+        self.do_finish = True
+
+
+async def test_workflow_current_build_id_appropriately_set(client: Client):
```

Comment:
```suggestion
async def test_workflow_current_build_id_appropriately_set(client: Client):
    if env.supports_time_skipping:
        pytest.skip("Java test server does not support worker versioning")
    if not await worker_versioning_enabled(client):
        pytest.skip("This server does not have worker versioning enabled")
```

Ignore this test when on time-skipping server or on older server

Reply:
Worker versioning isn't actually being used here - just the fields about build id which should work in the test server, right?

Reply:
I thought I saw a CI error when we run these tests against time skipping. But now I'm seeing some other update error.

Reply:
I mean it shouldn't need to be skipped at all, the propagation of this data should also work on test server

Reply:
🤷 Something else is breaking CI, unsure what, I can debug if necessary

Reply:
Yeah, I don't know what's up with that. It's the test I wrote back when I added update but it was working perfectly fine but seems consistently broken here, but nothing here should affect it...

Reply:
Huh, weird. I did manage to repro it once locally. I think it just flakes very rarely. Looks like there might be some race where occasionally the update handler is not registered on the first task. I'm not totally sure how that could be.

Reply:
I agree, something confusing is happening here. From the code, surely the start workflow task should run before the update attempt. I can't see from the code how that could be either. I wonder in the case where you repro if somehow the history has the update _before_ the start?

I do think it's quite an important find and I think we should definitely dig into it, but I understand if you don't want it holding this up. You can either open an issue and skip the test (or that part at least), or feel free to investigate inline if you don't mind it holding this up. I will ask internally if anyone knows of a case today where a start + update could result in an update + start in history.

Reply:
Ok, we figured this out. We intentionally order updates before start, so in the case where your client command start-then-updates before the task is delivered, update will run first. And we don't have a buffer-update-for-life-of-task like Go, Java, and TS do. We will discuss internally, but in the meantime this test is a bit flawed in assuming updates always come after start.